### ISSUE TITLE
Component changes not reflected in Ionic Serve

### ISSUE THREAD

**Issue by t4tapas** [ionic-team/ionic-cli#2847]
**Description:**
The changes done specifically in any on my Ionic custom Components are not loading in Ionic Serve. I know this issue is posted before and tried everything but the issue is still is there. 
I have created created few Ionic custom Component as
`$ ionic generate component MyComponent`
It was working just fine for last 1 month and there was no changes to those component TS files.
Few days back I hade to change some code in one of the Component TS file.

A method inside the Component TS was calling another method of a Provider. I changed the number of parameters in the signature of the Provider from 2 to 1 and changed the Component method accordingly. Its still showing the error that my component method has 2 parameter and trying to calling my Provider method with 1 parameter.

Any other changes done in Pages or Provider are fine. Just the component TS file changes are not reflected.

**Steps to Reproduce:**
Tried runing in browser and emulators.
Tried `npm run build`

**Output:**
TypeScript error for the old code.

**My `ionic info`:**
```
@ionic/cli-utils  : 1.15.2
ionic (Ionic CLI) : 3.15.2
```
global packages:

`cordova (Cordova CLI) : 7.1.0`
local packages:

```
@ionic/app-scripts : 3.0.1
Cordova Platforms  : android 6.3.0 ios 4.5.2
Ionic Framework    : ionic-angular 3.6.0
```
System:

```
ios-deploy : 1.9.1
ios-sim    : 5.0.13
Node       : v6.11.2
npm        : 3.10.10
OS         : macOS Sierra
Xcode      : Xcode 9.0.1 Build version 9A1004
```

**Comment by olivier-po** [comment]
I'm observing the same issue, but only with the html template. I need to quit the serve process and serve again to get the changes in the browser.

**Comment by amerzad** [comment]
@olivier-po 
It was happening in both HTML and TS, but when deleting all the storage from Chrome Development Tools the HTML started to reflect changes, but the TS file changes are still not reflected.
I downgraded to 3.13.0 and it didn't work, so it must be something other than ionic-cli.
I checked the source file in the hard disk in www/build/main.js, and it is not regenerated, so it is not related to browser at all.
It must be somewhere in the dependencies of ionic-cli. That is a tough thing to track.

**Comment by tskweres** [comment]
Same problem, putting new @Input() variables in my components and it's throwing errors saying they don't exist

**Comment by raxkaynan** [comment]
Also have the same problem. I use ionic-cli 3.19.0.

**Comment by amerzad** [comment]
@raxkaynan It is working now. I just followed the steps on this link:
https://angular-meteor.com/tutorials/whatsapp2/ionic/setup
And my code comes to work again with live reload.
I think it was a mistake in webpack-merge plugin I used to customize ionic webpack.config.js file.

**Comment by thielCole** [comment]
I've been seeing this issue to. Every serve it is having trouble updating changes. When I re-save the file though, it seems to catch up to the current version.
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.19.0
    ionic (Ionic CLI) : 3.19.0

global packages:

    cordova (Cordova CLI) : 7.1.0

local packages:

    @ionic/app-scripts : 3.1.2
    Cordova Platforms  : android 6.3.0 ios 4.5.2
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 25.2.5
    ios-deploy        : 1.9.2
    ios-sim           : 5.0.8
    Node              : v8.6.0
    npm               : 5.5.1
    OS                : macOS High Sierra
    Xcode             : Xcode 9.2 Build version 9C40b

Environment Variables:

    ANDROID_HOME : /Users/Cole/Library/Android/sdk

**Comment by laconstituante** [comment]
Any Solution for this issue? I am seeing this problem too with ionic 3.19.1
Initially thought it was because of the save all feature I use under vs code. but this seems to be the case event when saving a single file, html or ts.
the only work around is to stop the server and serve again which takes lots of time

**Comment by mydoal** [comment]
Ping. Any updates?

**Comment by pjhartin** [comment]
Same issue here, can be quite frustrating.

**Comment by jpercas** [comment]
I have the same error, is there no answer? I'am modify code but doesn't reflect the changes in emulators or browser.

**Comment by Rakonda** [comment]
Same error here

**Comment by yajuve** [comment]
Same problem

**Comment by jnovaes** [comment]
The only thing that works for me, after check out node/npm installs, was:
![captura de tela 2018-06-03 as 13 35 02](https://user-images.githubusercontent.com/4122376/40888837-8a2279e6-6733-11e8-98b3-65054d52f12b.png)

**Comment by erperejildo** [comment]
@jnovaes I tried that but still not working

**Comment by OutsourceNowZpos** [comment]
I been having this issue for over a month now, I will save the changes again and it will work in the browser but once I build those changes are not reflective at all. Specific components for me. I go back to browser my changes are giving errors again like method doesn't exist in my ts, save either html or ts and starts working again. But build android no changes at all.

**Comment by slimlime** [comment]
Yup same here. Issue for months now and troubleshooted with those checkboxes steps. Inconsistent loads which cause errors where there shouldn't be.

**Comment by nazzal88** [comment]
I have the same error, is there no answer?

**Comment by AMRALAA10** [comment]
😞 have the same problem too

**Comment by en-sevilla** [comment]
any response from IONIC ? Please, this bug (or whatever it is) is making us waste a lot of time and effort, please IONIC support, take 30 seconds to give us some feedback.

Thanks again.

**Comment by OutsourceNowZpos** [comment]
So do you guys have ionic pro configured in you project? Somehow the Cordova-plugin-ionic was responsible for my 2months as a failure at my company. Worth looking at that if you project has ionic pro configured. I simply Uninstalled it.

**Comment by andrewvmail** [comment]
Okay this worked guys thanks to 
https://github.com/ionic-team/ionic-app-scripts/pull/1420/commits/984ffe716d8012e90318fabef4bed817ff27c19e
```
vim node_modules/@ionic/app-scripts/dist/template.js    
# comment out the break in line 29
```

**Comment by richiglez** [comment]
> Okay this worked guys thanks to
> [ionic-team/ionic-app-scripts@984ffe7](https://github.com/ionic-team/ionic-app-scripts/commit/984ffe716d8012e90318fabef4bed817ff27c19e)
> 
> ```
> vim node_modules/@ionic/app-scripts/dist/template.js    
> # comment out the break in line 29
> ```

That did the trick for me. Awesome!

**Comment by themightysapien** [comment]
In inoic 4 beta what i did was ng run app:serve --poll=2000 to solve it. Dev server running on default angular port

**Comment by dwieeb** [comment]
@themightysapien Let's keep this issue for v3. In fact, I'll be moving it to the app-scripts repo.
